gum/arch-mips/gummipsrelocator.c

```diff
@@ -0,0 +1,530 @@
+/*
+ * Copyright (C) 2014-2015 Ole André Vadla Ravnås <ole.andre.ravnas@tillitech.com>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+/* Useful reference: C4.1 A64 instruction index by encoding */
+
+#include "gummipsrelocator.h"
+
+#include "gummemory.h"
+
+#define GUM_MAX_INPUT_INSN_COUNT (100)
+
+typedef struct _GumCodeGenCtx GumCodeGenCtx;
+
+struct _GumCodeGenCtx
+{
+  const cs_insn * insn;
+  cs_mips * detail;
+
+  const cs_insn * delay_slot_insn;
+  cs_mips * delay_slot_detail;
+
+  GumMipsWriter * output;
+};
+
+static gboolean gum_mips_has_delay_slot (const cs_insn * insn);
+
+void
+gum_mips_relocator_init (GumMipsRelocator * relocator,
+                          gconstpointer input_code,
+                          GumMipsWriter * output)
+{
+  cs_err err;
+#ifdef G_LITTLE_ENDIAN
```

Comment:
Actually, we can get rid of this whole `#ifdef` by just using `GUM_DEFAULT_CS_MODE` that we already defined.